src/index.js

```diff
@@ -151,6 +144,31 @@ export default class ScrollBehavior {
     });
   }
 
+  _setScrollRestoration = () => {
+    if (this._hasSetScrollRestoration) {
+      // It's possible that the `pageshow` event occurs after we initialise
+      return;
+    }
+    if (
+      'scrollRestoration' in window.history &&
+      // Unfortunately, Safari on iOS freezes for 2-6s after the user swipes to
+      // navigate through history with scrollRestoration being 'manual', so we
+      // need to detect this browser and exclude it from the following code
+      // until this bug is fixed by Apple.
+      !isMobileSafari()
+    ) {
+      this._oldScrollRestoration = window.history.scrollRestoration;
+      try {
+        window.history.scrollRestoration = 'manual';
+      } catch (e) {
+        this._oldScrollRestoration = null;
+      }
+    } else {
+      this._oldScrollRestoration = null;
+    }
+    this._hasSetScrollRestoration = true;
```

Comment:
there's probably no reason to set this except in the case above where the `window.history.scrollRestoration` assignment succeeded; there's nothing to restore in those cases, and even clearly `_oldScrollRestoration` is irrelevant

Reply:
or maybe better yet, just merge `_oldScrollRestoration` and `_hasSetScrollRestoration`; the condition can just be `_oldScrollRestoration != null`